src/draive/conversation/realtime/types.py

```
@@ -0,0 +1,21 @@
+from draive.tools import Toolbox
+
+__all__ = ("RealtimeConversationStarting",)
+
+
+@runtime_checkable
+class RealtimeConversationStarting(Protocol):
+    async def __call__(
+        self,
+        *,
+        instruction: Instruction | None,
+        input: AsyncIterator[ConversationStreamElement],  # noqa: A002
```

Pylint (3.3.7)

[warning] Redefining built-in 'input'

(W0622)